File: sync_static_site.py
import json
import shutil
from pathlib import Path

import server


BASE_DIR = Path(__file__).resolve().parent
DOCS_DIR = BASE_DIR / "docs"
DOCS_DIR.mkdir(exist_ok=True)

STATIC_FILES = [
    "index.html",
    "app.js",
    "styles.css",
    "service-worker.js",
    "manifest.webmanifest",
    "icon.svg",
    "icon-192.png",
    "icon-512.png",
    "apple-touch-icon.png",
]


def sync_static_files() -> None:
    for name in STATIC_FILES:
        shutil.copy2(BASE_DIR / name, DOCS_DIR / name)

    (DOCS_DIR / ".nojekyll").write_text("", encoding="utf-8")


def write_briefing() -> None:
    payload = server.build_digest()
    target = DOCS_DIR / "briefing.json"
    target.write_text(json.dumps(payload, ensure_ascii=False, indent=2), encoding="utf-8")


def main() -> None:
    sync_static_files()
    write_briefing()


if __name__ == "__main__":
    main()

File: sync_static_site.js
import fs from "node:fs";
import path from "node:path";
import { spawnSync } from "node:child_process";

const BASE_DIR = process.cwd();
const DOCS_DIR = path.join(BASE_DIR, "docs");

const STATIC_FILES = [
  "index.html",
  "app.js",
  "styles.css",
  "service-worker.js",
  "manifest.webmanifest",
  "icon.svg",
  "icon-192.png",
  "icon-512.png",
  "apple-touch-icon.png",
  "loading-preview.html",
  "content.js",
];

function copyStaticFiles() {
  fs.mkdirSync(DOCS_DIR, { recursive: true });
  for (const file of STATIC_FILES) {
    fs.copyFileSync(path.join(BASE_DIR, file), path.join(DOCS_DIR, file));
  }
  fs.writeFileSync(path.join(DOCS_DIR, ".nojekyll"), "", "utf8");
}

const build = spawnSync(process.execPath, [path.join(BASE_DIR, "build_content.mjs")], {
  stdio: "inherit",
});

if (build.status !== 0) {
  process.exit(build.status ?? 1);
}

copyStaticFiles();
